temporian/implementation/numpy/data/dtype_normalization.py

```diff
@@ -177,8 +180,10 @@ def normalize_timestamps(
         values = np.array(values)
 
     # values is represented as a number. Cast to float64.
-    if values.dtype.type in [np.float32, np.int64, np.int32]:
-        values = values.astype(np.float64)
+    if np.issubdtype(values.dtype, np.integer) or np.issubdtype(
+        values.dtype, np.floating
+    ):
+        values = values.astype(np.float64, copy=False)
```

Comment:
I would do a copy here.

Reply:
Do we also want to copy `float64` inputs to decouple from user's data?